### ISSUE TITLE
Investigate why XML export/import tests not executed

### ISSUE THREAD

**Issue by Siedlerchr** [JabRef/jabref#2050]
Noticed in #1942

**Comment by Siedlerchr** [comment]
Hm. Running the tests in eclipse they are all executed succesfully. The Paramterized tests work.
@boceckts  Could you please check that in intellij, too?
`BibTeXMLImporterTestFiles` should execute 40 tests

**Comment by boceckts** [comment]
No test gets executed in IntelliJ when executing the test file directly. I added a sysout to see the full path of the test files and it seems like I found the issue. 
```Java
@Parameters(name = "{0}")
    public static List<Path> files() throws Exception {
        try (Stream<Path> stream = Files.list(Paths.get(BibTeXMLImporterTest.class.getResource("").toURI()))) {
            stream.forEach(System.out::println);
            return stream.filter(n -> n.getFileName().toString().startsWith("BibTeXMLImporterTest")
                    && n.getFileName().toString().endsWith(".xml")).collect(Collectors.toList());
        }
    }
```
console output
``` Java
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\BiblioscapeImporterTest.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\BiblioscapeImporterTestFiles.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\BiblioscapeImporterTestTypes.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\BibtexImporterTest.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\BibTeXMLImporterTest.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\BibTeXMLImporterTestFiles.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\BibTeXMLImporterTestTypes.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\BibtexParserTest.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\CopacImporterTest.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\CopacImporterTestFiles.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\CustomImporterTest.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\EndnoteImporterTest.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\FreeCiteImporterTest.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\ImportFormatTest.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\InspecImportTest.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\IsiImporterTest.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\MedlineImporterTest.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\MedlineImporterTestFiles.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\MedlinePlainImporterTest.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\ModsImporterTestFiles.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\MsBibImporterTest.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\MsBibImporterTestfiles.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\OvidImporterTest.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\PdfContentImporterTest.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\PdfContentImporterTestFiles.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\PdfXmpImporterTest.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\RepecNepImporterTest.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\RISImporterTest.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\RISImporterTestFiles.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\SilverPlatterImporterTest.class
C:\Users\Tobias\git\jabref\build\classes\test\net\sf\jabref\logic\importer\fileformat\SilverPlatterImporterTestNotRecognized.class
```
for me this prints out all the class files of the importer, which means it is the wrong directory. Strangely this differs when using eclipse where @tschechlovdev tested the same and was in the directory of the test files.

**Comment by Siedlerchr** [comment]
Hm, looks like the class dir is not right:

`BibTeXMLImporterTest.class.getResource("")`
Must be` BibTeXMLImporterTestFiles.class.getResource("")`

**Comment by Siedlerchr** [comment]
I think I found the answer: Is a setting:
http://stackoverflow.com/questions/29607053/different-classloader-paths-for-eclipse-and-intellij

**Comment by tschechlovdev** [comment]
@Siedlerchr But how does this solve the problem on codecov?

**Comment by Siedlerchr** [comment]
At least that is the way to make it work in intellij, too.
@tschechlovdev  No idea. I could  not find any documentation at codecov regarding parameterized junit tests.

**Comment by tschechlovdev** [comment]
Is there a reason why ``getRessource("")`` and not ``getRessource(PATH_TO_FILE)`` was used? If using the explicit path  it seems to work.

**Comment by Siedlerchr** [comment]
It was probably because we wanted to get the folder of the resource and not the file itself. 
Because of the parameterized test. Do the parameterized tests execute all?

When I understand this correctly, the whole path should work:
`The elements of the stream are Path objects that are obtained as if by resolving the name of the directory entry against dir`
https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#list-java.nio.file.Path-

**Comment by tschechlovdev** [comment]
Fixed in #2121.
